### PR TITLE
common: add KeyValueModel

### PR DESCRIPTION
**Pull request by bajtos**
This is the initial REST API for key-value models.

```
GET /my-models/key
PUT /my-models/key
PUT /my-models/key?ttl=123
PUT /my-models/key/expire
```

@ritch @superkhau please review

Connect #2525
Requires https://github.com/strongloop/loopback-datasource-juggler/pull/1033
Requires https://github.com/strongloop/loopback-datasource-juggler/pull/1039

### CODE REVIEW

**Comment by bajtos** [comment]
There are few test failures, I am aware of them and will investigate tomorrow. For now, please focus on reviewing the user-facing API.

File: common/models/key-value-model.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  // TODO add api docs
+  KeyValueModel.get = function(key, options, callback) {
+    throwNotAttached(this.modelName, 'get');
+  };
+
+  // TODO add api docs
+  KeyValueModel.set = function(key, value, options, callback) {
+    throwNotAttached(this.modelName, 'set');
+  };
+  //
```
Comment (superkhau) [inline common/models/key-value-model.js]: remove //

**Comment by superkhau** [comment]
LGTM

**Comment by ritch** [comment]
LGTM

**Pull request merged**